core/src/main/java/io/keen/client/java/KeenClient.java

```diff
@@ -160,6 +161,9 @@ public void addEvent(KeenProject project, String eventCollection, Map<String, Ob
         KeenProject useProject = (project == null ? defaultProject : project);
 
         try {
+            if (eventCollection.contains(" ")){
+                eventCollection = URLEncoder.encode(eventCollection, "UTF-8").replace("+", "%20");
```

Comment:
There are a couple issues in this line:

1. AFAICT `URLEncoder.encode` will replace `+` with `%2B`, so the `.replace("+", "%20")` will always be a no-op.

2. Implicitly performing URL encoding *only in the case of the collection name containing a space* seems unintuitive. That means if I call this method with `eventCollection` equal to `%completed` then it will be submitted to Keen as `%completed` but if I call the method with `% completed` it will be submitted to Keen as `%25%20completed`.